Formula/ruby@1.8.7.rb

```diff
@@ -0,0 +1,205 @@
+  homepage "https://www.ruby-lang.org/"
+  url "https://cache.ruby-lang.org/pub/ruby/1.8/ruby-1.8.7-p374.tar.bz2"
+  sha256 "b4e34703137f7bfb8761c4ea474f7438d6ccf440b3d35f39cc5e4d4e239c07e3"
+  revision 2
+
+  bottle do
+    sha256 "bc89a74b22eec03454e42a37ce12df2fd31765e3ed9f87673fb3d073f0d3db05" => :sierra
+    sha256 "85ffdf739e1f4f74979e59616933e57a3a384bfc2214277b34120295ab0a695d" => :el_capitan
+    sha256 "f40e11d81cdae7051600b72ea5511b9995420618a13b65be439f6978e634d8c5" => :yosemite
+  end
+
+  option :universal
+  option "without-suffix", "Do not suffix commands with '187' (will become keg-only)"
+  option "with-doc", "Install documentation"
+  option "with-tcltk", "Install with Tcl/Tk support"
+
+  provided_by_osx = [:mountain_lion, :lion, :snow_leopard].include? MacOS.version.to_sym
+  if provided_by_osx && (build.without? "suffix")
+    keg_only :provided_by_osx
+  end
+
+  depends_on "pkg-config" => :build
+  depends_on "readline" => :recommended
+  depends_on "gdbm" => :optional
+  depends_on "openssl"
+  depends_on :x11 if build.with? "tcltk"
+
+  resource "rubygems" do
+    url "https://rubygems.org/rubygems/rubygems-2.5.0.tgz"
+    sha256 "a39e95acb86f52fdca38b1484b2eed61a7f76551119079d7a9f8c67c5be78831"
+  end
+
+  def install
+    # Compilation with `superenv` breaks because the Ruby build system sets
+    # `RUBYLIB` and `RUBYOPT` and this disturbs the wrappers for `cc` and
+    # `clang` that are provided by Homebrew and are implemented in Ruby. We
+    # prefix those invocations with `env` to clean the environment for us.
+    scrub_env = "/usr/bin/env RUBYLIB= RUBYOPT="
+    inreplace "mkconfig.rb", "    if /^prefix$/ =~ name\n",
+              <<-EOS.undent.gsub(/^/, "    ")
+                # `env` removes stuff that will break `superenv`.
+                if %w[CC CPP LDSHARED LIBRUBY_LDSHARED].include?(name)
+                  val = val.sub("\\"", "\\"#{scrub_env} ")
+                end
+                if /^prefix$/ =~ name
+              EOS
+
+    args = %W[
+      --prefix=#{prefix}
+      --enable-shared
+      --with-sitedir=#{HOMEBREW_PREFIX}/lib/ruby/site_ruby
+      --with-vendordir=#{HOMEBREW_PREFIX}/lib/ruby/vendor_ruby
+    ]
+
+    if build.universal?
+      ENV.universal_binary
+      args << "--with-arch=#{Hardware::CPU.universal_archs.join(",")}"
+    end
+
+    args << "--program-suffix=#{program_suffix}" if build.with? "suffix"
+    args << "--enable-install-doc" if build.with? "doc"
+
+    rm_r "ext/tk" if build.without? "tcltk"
+
+    system "./configure", *args
+
+    # Ruby has been configured to look in the HOMEBREW_PREFIX for the
+    # sitedir and vendordir directories; however we don't actually want to create
+    # them during the install.
+    #
+    # These directories are empty on install; sitedir is used for non-rubygems
+    # third party libraries, and vendordir is used for packager-provided libraries.
+    inreplace "instruby.rb" do |s|
```

Comment:
For the sake of just porting this from homebrew/versions I think it's fine as-is.